packages/deployments/contracts/contracts/core/connext/helpers/RelayerProxyHub.sol

```diff
@@ -139,21 +154,59 @@ contract RelayerProxyHub is RelayerProxy {
 
   IRootManager public rootManager;
   uint256 public propagateCooldown;
+  uint256 public proposeAggregateRootCooldown;
   // Timestamp of the last time the job was worked.
   uint256 public lastPropagateAt;
+  uint256 public lastProposeAggregateRootAt;
+
+  enum AutonolasPriorityFunction {
+    Propagate,
+    ProcessFromRoot,
+    ProposeAggregateRoot,
+    FinalizeAndPropagate
+  }
+
+  // number between 0 and 10 to determine priority that Autonolas has for jobs
+  // 0 is disabled, 10 will work for every block
+  mapping(AutonolasPriorityFunction => uint8) public autonolasPriority;
+
   mapping(uint32 => mapping(bytes32 => bool)) public processedRootMessages;
   mapping(uint32 => address) public hubConnectors;
 
   // ============ Events ============
   event RootManagerChanged(address rootManager, address oldRootManager);
   event PropagateCooldownChanged(uint256 propagateCooldown, uint256 oldPropagateCooldown);
   event HubConnectorChanged(address hubConnector, address oldHubConnector, uint32 chain);
+  /**
+   * @notice Emitted when Autonolas priority is updated by admin
+   * @param updated New Autonolas priority in the contract
+   * @param previous Old Autonolas priority in the contract
+   */
+  event AutonolasPriorityChanged(AutonolasPriorityFunction fn, uint8 updated, uint8 previous);
 
   // ============ Errors ============
   error RelayerProxyHub__propagateCooledDown_notCooledDown(uint256 timestamp, uint256 nextWorkable);
+  error RelayerProxyHub__finalizeAndPropagateCooledDown_notCooledDown(uint256 timestamp, uint256 nextWorkable);
+  error RelayerProxyHub__proposeAggregateRootCooledDown_notCooledDown(uint256 timestamp, uint256 nextWorkable);
   error RelayerProxyHub__processFromRoot_alreadyProcessed(uint32 chain, bytes32 l2Hash);
   error RelayerProxyHub__processFromRoot_noHubConnector(uint32 chain);
 
+  // ============ Modifiers ============
+  /**
+   * @notice Indicates if the job is workable by the sender. Takes into account the Autonolas priority.
+   * For example, if priority is 3, then sender will not be able to work on blocks 0, 1, 2, unless they are Autonolas.
+   * Priority 0 disables Autonolas priority completely."
+   * @param _sender The address of the caller
+   */
+  modifier isWorkableBySender(AutonolasPriorityFunction _function, address _sender) {
+    if (
+      _sender != autonolas && autonolasPriority[_function] != 0 && block.number % 10 <= autonolasPriority[_function] - 1
+    ) {
```

Comment:
I feel like the `_sender` condition is wrong here -- the rest of the conditions are relevant IFF its autonolas

Reply:
but im not sure I understand why some of these functions are restricted